src/interest-groups/join.js

```diff
@@ -1,3 +1,23 @@
+import {
+	hasInvalidOptionTypes,
+	isMissingRequiredOptions,
+	validateParam,
+} from '../utils/';
+import {
+	MAX_EXPIRATION,
+	REQUIRED_OPTS,
+} from './constants';
+import types from './types';
+
 export default function joinAdInterestGroup (options, expiry) {
+	validateParam(options, 'object');
+	validateParam(expiry, 'number');
+	isMissingRequiredOptions(options, REQUIRED_OPTS);
+	hasInvalidOptionTypes(options, types);
+
+	if (expiry > MAX_EXPIRATION) {
+		throw new Error(`'expiry' is set past the allowed maximum value. You must provide an expiration that is less than or equal to ${MAX_EXPIRATION} (or 30 days in milliseconds per day).`);
```

Comment:
its a little weird to me that you have the MAX_EXPIRATION as a variable but the error message has it hardcoded as "30 days in ms per day". To stay consistent I would just note that the max expiration is in ms no need to give the 30 days conversion.